### ISSUE TITLE
Crash On 1.21 With Latest Version

### ISSUE THREAD

**Issue by Cramram** [Origins-Tech/InventiveInventory#6]
I'm using Inventive Inventory 1.0.1 on Minecraft 1.21, and whenever I'm playing and going through shulker boxes, the game will crash. It then proceeds to save my world data, but not my player data, although I think that's just how Minecraft handles crashes. It happened numerous times, specifically while rapidly opening and closing shulker box and villager menus. This did not happen on Inventive Inventory 0.4.6, and stopped as soon as I removed it. Thanks in advance! I love this mod, it's by far my favorite sorting mod.

**Comment by Strobel273** [comment]
Hey Cramram,
thank you for your report and the kind words.

We weren't quite able to reproduce the issue. 
Could you please provide a more detailed way on how the crashes occured?
Did you play in singleplayer or on a server?

Thanks again,
Your Inventive Inventory Team

**Comment by Cramram** [comment]
Here is a video of it happening. But for more info, it would happen when quickly switching between different menus, such as villagers and shulker boxes. It would just cause the game to freeze, and eventually crash. The Minecraft crash report said that I needed more ram, but I had 10gb dedicated to the game.

https://github.com/user-attachments/assets/69dd2738-3735-4056-8aff-e7bddcdd6791

Sorry for my audio, was recording for a YouTube video.

**Comment by Strobel273** [comment]
Yeah that's strange. Could you please send us the crash report?

**Comment by Cramram** [comment]
Of course. Also, I found another bug while doing so. It also only started happening when I updated, but when shift clicking and dragging a lot of items into my inventory with the Inventory Tweaks mod, it shifts locked items around.
![bug report](https://github.com/user-attachments/assets/f27583b0-8300-48cd-b53e-71ee9f7ccabf)

**Comment by Strobel273** [comment]
I would need the crash log, sorry. If you don't know where it is located, go to your minecraft instance folder > logs > latest.log.

Regarding the shifts of locked items, using multiple inventory mods is always a little bit of risk and there's a high chance of incompatibility. Nevertheless, we will look into this and try to resolve this. We know of course that those mods provide very convenient ways to move items and manage the inventory that would be missed.

**Comment by Cramram** [comment]
Ah, sorry for misunderstanding, and apologies for not answering your question. It's a singleplayer world. I am using quite a few client side mods as well, but these issues didn't happen with the same exact mods before I updated Inventive Inventory.l
[latest.log](https://github.com/user-attachments/files/16931526/latest.log)
I hope this helps!

**Comment by LuoliiQaQ** [comment]
Hello, I know what caused your lack of response. This module is called itemscroller, a masa module made by Sakura. An update to this module adds a feature called one-click organizing, but its logic is very confusing. It can be seen in the video that there is an ender box in his backpack. In the original version, the ender box cannot be put into the shulker box, but itemscroller will keep trying to put the Ender box into the Shulker box, which causes an endless loop and sends data indefinitely. You can only wait for the author of itemscroller to update and fix this bug. The temporary workaround is to go to itemscroller in the mod menu and change the hotkey's R key to unspecified (press esc once), so this bug has nothing to do with the author's own mod.

**Comment by LuoliiQaQ** [comment]
It is recommended to close the lssues because the bug is caused by itemscroller

**Comment by Cramram** [comment]
Sorry, but I don't have anything called "itemscroller" installed, nor have I heard of it. You can check my log file, it's not there. I appreciate the help though.

**Comment by Strobel273** [comment]
We have uploaded a new version to github which doesn't directly deal with your issue but could have an impact as we improved the system behind quick moving and item pickup. Those features are in use in your video what we see from our perspective.
Please just try it out and provide feedback if anything has changed.
In your log file there wasn't anything connected to our mod and no error was thrown. It looks kind of suspicious at the end which could be caused by an endless loop as LuoliiQaQ stated. But we haven't seen the itemscroller mod either.
Also we couldn't reproduce the issue by now, are you absolutely sure everything works when you disable our mod?

If you can't build the mod by yourself i could send you the jar file on Discord. 
Just let us know.

**Comment by LuoliiQaQ** [comment]
I'm sorry, I recommend you try to remove all the mods, leave the author's mods and preposition  and try again, if there is a problem it must be what I said, it is constantly trying to put immovable items inside the shulker box, such as the Ender box
